### PR TITLE
Copy headers from original response when returning not modified.

### PR DESCRIPTION
**Pull request by croudet**
Otherwise headers added by filters are lost.

### CODE REVIEW

File: http-server-netty/src/main/java/io/micronaut/http/server/netty/types/files/FileTypeHandler.java [hunk 1]
```diff
@@ -157,8 +157,13 @@ protected void setDateHeader(MutableHttpResponse response) {
         headers.date(now);
     }
 
-    private FullHttpResponse notModified() {
+    private static void copyHeaders(NettyMutableHttpResponse<?> from, NettyMutableHttpResponse to) {
+        from.getHeaders().forEachValue((header, value) -> to.getHeaders().set(header, value));
```
Comment (graemerocher) [inline http-server-netty/src/main/java/io/micronaut/http/server/netty/types/files/FileTypeHandler.java:161]: I think `set(..)` will override the header for multi value headers and `add` should be used instead

Comment (croudet) [reply http-server-netty/src/main/java/io/micronaut/http/server/netty/types/files/FileTypeHandler.java]: Done.

**Comment by croudet** [comment]
Build failed:
```
Caused by:
        org.testcontainers.containers.ContainerFetchException: Can't get Docker image: 
```

**Comment by graemerocher** [comment]
Yes test containers is down because of https://status.quay.io/

**Comment by jameskleeh** [comment]
I don't think we should be copying all headers:
https://tools.ietf.org/html/rfc2616#section-10.3.5

> If the conditional GET used a strong cache validator (see Weak and Strong Validators), the response SHOULD NOT include other entity-headers. Otherwise (i.e., the conditional GET used a weak validator), the response MUST NOT include other entity-headers; this prevents inconsistencies between cached entity-bodies and updated headers.

https://tools.ietf.org/html/rfc2616#section-7.1

**Comment by croudet** [comment]
I have changed the copy function to exclude entity headers.

**Comment by ElArtista** [comment]
Well done!

**Pull request merged**